### DISCUSSION TITLE
Question about the license

### DISCUSSION THREAD

**Discussion by orf** [reloadware/reloadium#35] [Q&A] [answered]
Hey, cool project!

 I wanted to take a look at the source code but it seems that this is a closed source project. It would be quite good to document that somewhere on the website or within the project itself, as it's not clear at all.

I had a look through the binary, and it seems like you're distributing parts of the python standard library (urllib in particular) in it. I'm not an expert, but I think you need to [also include the PSF license?](https://wiki.python.org/moin/PythonSoftwareFoundationLicenseFaq#If_I_bundle_Python_with_my_application.2C_what_do_I_need_to_include_in_my_software_and.2For_printed_documentation.3F).

It also seems like portions of the [live server](https://github.com/tapio/live-server/blob/01cd7f970439d42537bbe13b4885befb816a0446/injected.html) project are bundled without the matching MIT license?

As a side note, are you going to open source reloadium at any point?

**Comment by dkrystki** [comment, marked as answer]
Hey Orf, thanks for submitting this.
My opensource licenses knowledge is very limited but as far I understand Apache 2.0 license is compatible with the licenses you mentioned.

About the project being partially closed source. It was fully open source at first (under the name smart-reloader back then) but I made a decision to make some parts closed sourced.
The reason behind it is that I don't want this project to be rebranded by big players like Microsoft without any or minimal recognition to reloadium. 
This is what happened to pydevd - an amazing opensource debugger for python that has been around for almost 10 years but barely anyone knows about it's existence even though almost everybody uses it everyday.
Microsoft made a thin wrapper around it and released under name `debugpy` and calls it Microsoft's "brand-new Python debugger" [0] which is part of VsCode now.
I understand that this is what opensource is about but there's absolutely no recognition for pydevd even though 99% of the project is pydevd.
I just don't want this project to become next "brand-new hot reloader by company X".

I am planning to opensource as much of the project as possible like the extensions and the extension manager so it is easy to write custom extension for libraries and frameworks.
PyCharm plugin is a big chunk of this project and it's fully opensource as well as VsCode plugin which is still in progress. 
I will keep the core closed source for now but I am open to suggestions how this could be handled differently.

[0] https://devblogs.microsoft.com/python/python-in-visual-studio-code-march-2020-release/

**Comment by youtux** [comment]
About not being open source, this is a big blocker for me as I am not able to audit the code before running it on my machine.
Another downside is that users can't compile it themselves, so it's unusable on Apple Silicon.

**Comment by adoxentor** [comment]
Please look at making it open source with a strong copyright licence. It will allow you to accept help and pull requests from the community. Looking at the source is extremely important to me, for security reasons. A project clone that uses open source but licenced code can be sued, and be companies will not risk it.
